src/corelibs/U2Core/src/datatype/primers/PrimerStatistics.cpp

```diff
@@ -57,6 +57,23 @@ QString PrimerStatistics::checkPcrPrimersPair(const QByteArray &forward, const Q
     return message;
 }
 
+double PrimerStatistics::getDeltaG(const QByteArray& sequence) {
+    CHECK(validate(sequence), Primer::INVALID_TM);
+
+    double freeEnergy = 0.0;
+    for (int i = 0; i < sequence.size() - 1; i++) {
+        QByteArray curArray;
+        curArray.append(sequence.at(i));
+        curArray.append(sequence.at(i + 1));
+        bool homologousRegionEnded = !BaseDimersFinder::ENERGY_MAP.contains(curArray);
+        if (!homologousRegionEnded) {
+            freeEnergy += BaseDimersFinder::ENERGY_MAP[curArray];
```

Comment:
Do not search twice in the map. This is not a free op in the hot loop. Re-use contains() result (replace contains with get, check for 0 energy).

Reply:
OK